src/basic/conf-files.c

```diff
@@ -163,16 +166,70 @@ int conf_files_list_strv(
                 return -ENOMEM;
 
         /* The entries in the array given by hashmap_get_strv() are still owned by the hashmap. */
-        files = strv_copy(sv);
-        if (!files)
-                return -ENOMEM;
 
-        typesafe_qsort(files, strv_length(files), base_cmp);
-        *ret = TAKE_PTR(files);
+        if (root && !FLAGS_SET(flags, CONF_FILES_BASENAME)) {
+                _cleanup_free_ char *root_abs = NULL;
+
+                r = path_make_absolute_cwd(root, &root_abs);
+                if (r < 0)
+                        return r;
 
+                path_simplify(root_abs);
+
+                n = 0;
+                STRV_FOREACH(p, sv) {
+                        char *q;
+
+                        q = path_join(root_abs, *p);
+                        if (!q)
+                                return -ENOMEM;
+
+                        r = strv_consume_with_size(&files, &n, q);
+                        if (r < 0)
+                                return r;
+                }
```

Comment:
Sorry for the churn, but now that I see this version, I think I liked the previous one better. It's better to have the root prefixing done by CHASE_PREFIX_ROOT then do it all ourselves in this function. Apologies for the churn.

Reply:
I'll leave the final judgement up to you on which version to go with.

Reply:
OK. No problem. I also prefer the previous version.